linux_os/guide/system/network/network-nftables/set_nftables_base_chain/ansible/shared.yml

```diff
@@ -0,0 +1,71 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+#Name of the table
+{{{ ansible_instantiate_variables("var_nftables_table") }}}
+#Familiy of the table 
+{{{ ansible_instantiate_variables("var_nftables_family") }}}
+#Number of base chains
+{{{ ansible_instantiate_variables("var_nftables_base_chain_number") }}}
+#Name(s) of base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_names") }}}
+#Type(s) of base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_types") }}}
+# Hooks for base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_hooks") }}}
+#Priority
+{{{ ansible_instantiate_variables("var_nftables_base_chain_priorities") }}}
+#Policy 
+{{{ ansible_instantiate_variables("var_nftables_base_chain_policies") }}}
+
+- name: Ensure Base Chains Exist for Nftables - Remove file for remediation
+  ansible.builtin.file:
+    path: ~/set_nftables_base_chain.sh
+    state: absent
+
+- name: Ensure Base Chains Exist for Nftables - Generation of script for remediation
+  ansible.builtin.copy:
+    dest: "~/set_nftables_base_chain.sh"
```

Comment:
This would create a file in the home directory of the user running the playbook, usually `root` in this case. I have some concerns about this. First, the root of the `home` dir is not the most appropriated place to store shell scripts. Also, some site policies may not allow creation of shell scripts locally. What would happen if the system is already using another `nftables` script? If a remediation is reasonable, it would be better to be defined and ensured by Ansible tasks instead of using the Ansible tasks to create a shell script and execute it.